### PR TITLE
EDM-288: changing the way we create ignition paths from GitConfigProviderSpec

### PR DESCRIPTION
**Pull request by tsorya**
EDM-288: changing the way we create ignition paths from GitConfigProviderSpec
    This should fix the issue when user provides path like /file.txt and we
    set in ignition path as "/" , from now on we will set path as was
    provided by user

### CODE REVIEW

File: internal/tasks/git_helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	When("the path is empty", func() {
+		It("returns an error", func() {
+			mfs := memfs.New()
+
+			_, err := ConvertFileSystemToIgnition(mfs, "")
+			Expect(err).To(HaveOccurred())
```
Comment (hexfusion) [inline internal/tasks/git_helpers_test.go:66]: :+1:

File: internal/tasks/git_helpers.go [hunk 2]
```diff
@@ -213,12 +213,12 @@ func ConvertFileSystemToIgnition(mfs billy.Filesystem, path string) (*config_lat
 		if err != nil {
 			return nil, fmt.Errorf("failed reading directory %s: %w", path, err)
 		}
-		err = addGitDirToIgnitionConfig(mfs, path, "/", files, &ignitionConfig)
+		err = addGitDirToIgnitionConfig(mfs, path, path, files, &ignitionConfig)
```
Comment (hexfusion) [inline internal/tasks/git_helpers.go:216]: :rocket:

File: internal/tasks/git_helpers.go [hunk 3]
```diff
@@ -213,12 +213,12 @@ func ConvertFileSystemToIgnition(mfs billy.Filesystem, path string) (*config_lat
...
 		if err != nil {
 			return nil, fmt.Errorf("failed converting directory %s to ignition: %w", path, err)
 		}
 	} else {
-		err = addGitFileToIgnitionConfig(mfs, path, "/", fileInfo, &ignitionConfig)
+		err = addGitFileToIgnitionConfig(mfs, path, filepath.Join("/", fileInfo.Name()), fileInfo, &ignitionConfig)
```
Comment (hexfusion) [inline internal/tasks/git_helpers.go:221]: lets make '/' a const? `RootPath`?

**Review by hexfusion — COMMENTED** [review]
one small nit, looks great

File: internal/tasks/git_helpers.go [hunk 4]
```diff
@@ -260,7 +270,7 @@ func addGitFileToIgnitionConfig(mfs billy.Filesystem, fullPath, ignPath string,
 	if err != nil {
 		return err
 	}
-
+	fmt.Println("BBBBBBBBBBBBBBBB", ignPath)
```
Comment (avishayt) [inline internal/tasks/git_helpers.go:273]: ?

**Comment by tsorya** [comment]
closed as was done as part of https://github.com/flightctl/flightctl/pull/495

**Pull request closed without merging**